File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
      
        for(int i = 0; i < n; i++) {
            long num = arr1[i] | arr2[i];
            String s = Long.toBinaryString(num);
            int zero = n - s.length();
            for(int j = 0; j < zero; j++) {
                s = '0' + s;
            }
            System.out.println(s);
            answer[i] = "";
            
            for(int j = 0; j < s.length(); j++) {
                if(s.charAt(j) == '1') {
                    answer[i] += "#";
                }
                else if(s.charAt(j) == '0'){
                    answer[i] += " ";
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    var answer = [];
    for(let i = 0; i < arr1.length; i++) {
        let num = arr1[i] | arr2[i];
        let bin = num.toString(2);
        let str = "";
        for(let j = 0; j < n - bin.length; j++) {
            str += "0";
        }
        bin = str + bin;
        console.log(bin);
        let find = "";
        for(let j = 0; j < bin.length; j++) {
            if(bin[j] === "1") find += "#";
            else if(bin[j] === "0")find += " ";
        }
        answer.push(find);
    }
    return answer;
}